From 430f012c5794873a84db81b93c5858ca96ea4559 Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Fri, 17 Mar 2017 11:14:00 +0100
Subject: [PATCH 21/49] build/files.c (processPackageFiles): Don't call
 generateBuildIDs for noarch.

We don't want to do build-id processing for noarch packages. It might be
that noarch packages do contain architecture depended files, but those are
already handled by processBinaryFiles.

This fixes the building of openbios in fedora.
https://bugzilla.redhat.com/show_bug.cgi?id=1433129

Signed-off-by: Mark Wielaard <mark@klomp.org>
(cherry picked from commit 5e82c7e8a8fc05195cdf622d0a120b9e70a9371b)
---
 build/files.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/build/files.c b/build/files.c
index 6c0ca39ef..9e58ae547 100644
--- a/build/files.c
+++ b/build/files.c
@@ -2378,10 +2378,14 @@ static rpmRC processPackageFiles(rpmSpec spec, rpmBuildPkgFlags pkgFlags,
 	goto exit;
 
 #if HAVE_LIBDW
-    if (generateBuildIDs (&fl) != 0) {
-	rpmlog(RPMLOG_ERR, _("Generating build-id links failed\n"));
-	fl.processingFailed = 1;
-	goto exit;
+    /* Check build-ids and add build-ids links for files to package list. */
+    const char *arch = headerGetString(pkg->header, RPMTAG_ARCH);
+    if (!rstreq(arch, "noarch")) {
+	if (generateBuildIDs (&fl) != 0) {
+	    rpmlog(RPMLOG_ERR, _("Generating build-id links failed\n"));
+	    fl.processingFailed = 1;
+	    goto exit;
+	}
     }
 #endif
 
-- 
2.13.2

